internal/store/interfaces.go

```diff
@@ -45,7 +45,10 @@ type ListenerConfig struct {
 	Hostname string
 	Port     int
 	Protocol string
-	TLS      bool
+
+	// TODO: is this field a sufficient reason to use TLSParams instead of TLS as the
+	// field name on Listener structs to avoid confusion?
+	TLS bool
```

Comment:
Highlighting this question.

Reply:
Yeah, probably, it'd make the term overloading a bit less heavy.

Reply:
Would changing this field to `TLSEnabled` or something be a viable alternative?

Reply:
I'd be fine with that, so long as it makes sense, I'm even ok collapsing it into an `Enabled` boolean within a larger `TLS` configuration structure -- the corresponding code that uses the field currently would need to be updated though. Currently these are all just used as in-memory representations of our configuration that map to the underlying consul config entries and k8s manifests, so it should be fine to make the change, and frankly I'd prefer to do that now rather than in the future if we have say, a persistent data store for some of this stuff.

Reply:
Think I was able to pretty cleanly refactor this into the TLSParams struct as an `Enabled bool` field.